src/Modal.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import ReactModal from 'react-modal';
+
+let rootSelector;
+
+export const setAppElement = (selector) => {
+    rootSelector = selector;
+    ReactModal.setAppElement(selector);
+};
+
+export class Modal extends Component {
+    rootElement = null;
+    previousScrollValue = null;
+    prevStyleValues = {};
+
+    componentDidMount() {
+        const { isOpen } = this.props;
+
+        this.rootElement = document.querySelector(rootSelector);
+
+        if (isOpen) {
+            const newStyleValues = this.computeElementValues();
+
+            this.rootElement.style.top = newStyleValues.top;
+            this.rootElement.style.left = newStyleValues.left;
+            this.rootElement.style.right = newStyleValues.right;
+            this.rootElement.style.position = newStyleValues.position;
+        }
+    }
+
+    getSnapshotBeforeUpdate(prevProps) {
+        const { isOpen: wasOpen } = prevProps;
+        const { isOpen } = this.props;
+
+        if (!wasOpen && isOpen) {
+            return this.computeElementValues();
+        }
+
+        if (wasOpen && !isOpen) {
+            return {
+                top: this.prevStyleValues.top,
+                left: this.prevStyleValues.left,
+                right: this.prevStyleValues.right,
+                bottom: this.prevStyleValues.bottom,
+                position: this.prevStyleValues.position,
+            };
```

Comment:
We only want to return a new snapshot when the prop `isOpen` changes. Since there are many other props passing through this component to the `react-modal`'s one, I'm only returning a snapshot when the specific prop `isOpen` changes.

This doesn't actually takes effect since we're checking again if it was the `isOpen` prop that changed in `componentDidUpdate` but it avoids the case where a snapshot would be passed with no intention of using it.

Hope I explained it correctly :)

Reply:
My bad, though you were talking about the `return null`. Just changed it to `return this.prevStyleValues` :+1: